### PR TITLE
Update global data types to use the new XML representation

### PR DESCRIPTION
**Pull request by hicklin**
This PR 
- updates the definitions for global data types in `global-structs.xml` and `semantic-tag-namespace-enums.xml`
- regenerates code via zap
- fixes the SDK code that used these types as they where previously defined

### CODE REVIEW

File: src/app/clusters/service-area-server/service-area-cluster-objects.h
Comment (bzbarsky-apple) [inline src/app/clusters/service-area-server/service-area-cluster-objects.h:43]: While you are here, just using `NullNullable` here would be shorter, but maybe we want to make it clear which things we are leaving null?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/clusters/service-area-server/service-area-cluster-objects.h
Comment (hicklin) [reply src/app/clusters/service-area-server/service-area-cluster-objects.h:43]: I'm going to move this note the [PR that's adding a builder](https://github.com/project-chip/connectedhomeip/pull/34683/files) since there are a lot of similar changes being made there.

**Review by tehampson — APPROVED** [review]

File: src/app/clusters/service-area-server/service-area-server.cpp
Comment (hicklin) [reply src/app/clusters/service-area-server/service-area-server.cpp:573]: The patch has same error in it. I'm going to do it manually. However, I'm going to put in on a separate branch cause I have a lot of PRs riding on this one.

Comment (hicklin) [reply src/app/clusters/service-area-server/service-area-server.cpp:573]: I made the change [here](https://github.com/project-chip/connectedhomeip/pull/34739).

File: src/app/clusters/service-area-server/service-area-cluster-objects.h
Comment (hicklin) [reply src/app/clusters/service-area-server/service-area-cluster-objects.h:43]: Added the note [here](https://github.com/project-chip/connectedhomeip/pull/34683#discussion_r1699742228).

File: src/app/clusters/service-area-server/service-area-server.cpp
Comment (tehampson) [reply src/app/clusters/service-area-server/service-area-server.cpp:573]: Discussed offline I will merge these changes after this PR merges

**Pull request merged**
